classes/Product.php

```diff
@@ -1578,16 +1578,21 @@ public function updateAttribute($id_product_attribute, $wholesale_price, $price,
         $combination = new Combination($id_product_attribute);
 
         if (!$update_all_fields) {
-            $combination->setFieldsToUpdate(array(
-                'price' => !is_null($price),
-                'wholesale_price' => !is_null($wholesale_price),
-                'ecotax' => !is_null($ecotax),
-                'weight' => !is_null($weight),
-                'unit_price_impact' => !is_null($unit),
-                'default_on' => !is_null($default),
-                'minimal_quantity' => !is_null($minimal_quantity),
-                'available_date' => !is_null($available_date),
-            ));
+            $fieldsToUpdate = compact(
+                'price',
+                'wholesale_price',
+                'ecotax',
+                'weight',
+                'unit_price_impact',
+                'default_on',
+                'minimal_quantity',
+                'available_date'
+            );
+            $fieldsToUpdate = array_filter(
+                $fieldsToUpdate,
+                create_function('$value', 'return isset($value);')
```

Comment:
`create_function(...)` is deprecated from PHP 7.2, using it is discouraged. There another way to define a anonymous function: http://php.net/manual/en/functions.anonymous.php

Reply:
Private function then. Because anonymous functions are not supported before php 5.3

Reply:
I'll make a private function ;)